packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,22 +1,66 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
 
 export type HeaderProps = {
+	productId: number;
 	title: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+const DEFAULT_PRODUCT_NAME = 'AUTO-DRAFT';
+
+export function Header( { productId, title }: HeaderProps ) {
+	const { isProductLocked, isSaving } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord } = select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				isProductLocked: isPostSavingLocked(),
+				isSaving: isSavingEntityRecord(
+					'postType',
+					'product',
+					productId
+				),
+			};
+		},
+		[ productId ]
+	);
+
+	const isDisabled = isProductLocked || isSaving;
+
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	function handleSave() {
+		saveEditedEntityRecord( 'postType', 'product', productId );
+	}
+
 	return (
 		<div
 			className="woocommerce-product-header"
 			role="region"
 			aria-label={ __( 'Product Editor top bar.', 'woocommerce' ) }
 			tabIndex={ -1 }
 		>
-			<h1 className="woocommerce-product-header__title">{ title }</h1>
+			<h1 className="woocommerce-product-header__title">
+				{ title === DEFAULT_PRODUCT_NAME
+					? __( 'Add new product', 'woocommerce' )
+					: title }
+			</h1>
+
+			<div className="woocommerce-product-header__actions">
+				<Button
+					onClick={ handleSave }
+					variant="primary"
+					isBusy={ isSaving }
+					disabled={ isDisabled }
+				>
+					{ __( 'Save', 'woocommerce' ) }
```

Comment:
I'm seeing in the new design that this is `Add` when the product has not yet been created yet (auto-draft).  Okay with keeping this for a follow-up though if you prefer.

Reply:
done here -> [d28c40b](https://github.com/woocommerce/woocommerce/pull/37152/commits/d28c40b8e0bc18436124fcd3e87f6d5cbd742939) Thanks @joshuatf